### ISSUE TITLE
heuristics 4 and 10 static and not intelligent 

### ISSUE THREAD

**Issue by Oussamab21** [enferas/Mgh-game#6]
if we look at heuristic 4 and 10 they are too static and it 's intelligent so it would be better if we define a dynamic function for them : 
my idea is 
the importance or the value of the position depends of multiple criteria like 
c1/ the part of the game (first ,second ,last)
c2/ the number of opponent piece neighbors
c3/the number of path that leads to that position 
c4/ the number of triples that this move will prevent 

part 1: so  I thought in the first part we make this score :
c2 is 40% of the total score of the postion 
c3 is 40% of the total one since in this part there is no a lot of peices on the board and if so the other heuristics will prevent the tirple that is why the c4 is 20% in this part of the game 

part 2: there is a lot of pieces on the board so : 
c2: is up to 60%  ,c3 is 20% , c4 is 20%

part3 :flaying:
c2:60% c4:40%  c3:   0% (since it is flaying part)

**Comment by enferas** [comment]
I don't think we need c1 because there are heuristic for the first part and another for the second part.
in addition to that c4 will be 0 all the time. because if there are any triple the order will not arrive to this situation because there are another heuristic to solve that with a bigger priority.

c2 and c3 are good but be careful to put the piece in a dead place. we don't need to put a piece and cannot move it in the second part.
